trin-core/src/utils/bytes.rs

```diff
@@ -27,6 +27,18 @@ pub fn random_32byte_array(leading_bit_zeros: u8) -> [u8; 32] {
     bytes
 }
 
+/// Encode hex with 0x prefix
+pub fn hex_encode<T: AsRef<[u8]>>(data: T) -> String {
+    format!("0x{}", hex::encode(data))
+}
+/// Decode hex with 0x prefix
+pub fn hex_decode(data: &str) -> Result<Vec<u8>, hex::FromHexError> {
+    match &data[..2] {
+        "0x" => hex::decode(&data[2..]),
+        _ => Err(hex::FromHexError::InvalidHexCharacter { c: 'x', index: 0 }),
```

Comment:
I'm a bit confused by this error message. Isn't it saying basically that it found an `x` at index `0`? Which doesn't really correspond with the match statement. But, also I'm not sure if it matters all that much. Is there any reason to maintain the `FromHexError` here? Using `anyhow` here might be a better solution

Reply:
Yeah, hence my:
> plus a little hacky error if it's not there.

> Is there any reason to maintain the `FromHexError` here? Using `anyhow` here might be a better solution

No good reason. It was really just convenience and my lack of anyhow familiarity. I'll dig in to fix it, because it will bother me too, someday...